doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,254 @@
+    - Slow metadata lookup
+    - Space overhead (live and unused data co-exist)
+
+- In-place update store
+  Conventional filsystems such as ext4, xfs have an in-place update manner. This type of 
+  the store overwrites data and metadta at a fixed offset depending on a file.
+
+  - Pros
+    - No host-side GC is required
+    - Fast lookup
+    - No space overhead
+  - Cons
+    - More writes occur to record the data (metadata and data section are separated)
+    - It cannot support transaction. WAL is required to support transactions
+    - Give more burdens on SSDs due to device-level GC
+
+Motivation
+----------
+
+In modern distributed storage systems, a server node can be equipped with multiple 
+NVMe storage devices. In fact, ten or more devices could be attached on a server.
+As a result, CPU resources to acheive NVMe SSDs's performance is much higher than before
+in terms of IOPS (I/O per seconds).
+To fully exploit NVMe's performance, now we should consider host-side overhead that 
+occurs due to high CPU consumption. In short, minimizing host-side CPU consumption
+caused by storage processing is crucial.
+
+From the perspective of CPU consumption at the store-level, the way that mitigates
+resource usage is minimizing internel operation while processing I/O.
+Conventional log-based store has a benefit of flash friendly data layout. 
+However, Key-value DB based on LSM tree causes host-side I/O amplification 
+including compaction. In addition, Log-structured store also produces unpredictable
+I/O by garbage collection.
+
+The ideal store would issue the same number of I/Os as requested by the user to the store 
+device without I/O amplification. However, it is impossible to store the data via
+only a single write because the store in Ceph requires supporting transaction which
+requires ACID. Therefore, we intent to minimize I/O amplification while supporing
+transactions by using a hybrid update scheme depending data types.
+
+
+Keyidea
+-------
+
+Out-of-place update scheme: | SB | Object metadata | Journal |
+In-place update scheme: | SB | Object metadata | Allocation bitmap | Data blocks |
+
+We are probably able to mitigate all the cons of the in-place update store 
+combining out-of-place benefit as follows
+- Use pre-allocation technique to reduce # of writes
+- Ordered-mode-like with hybrid update scheme utilizing NVMe feature + help from 
+  replica on recovery to minimize I/O amplification with supporting transaction
+- High-end NVMe SSD has enuough powers to handle more works
+- There is enough program-earse cycle limit [1]
+- Depening on data type, some data does not induce many overheads caused by GC
+
+Therefore, we propose PoseidonStore which makes use of different update strategy 
+to minimize CPU consumption and I/O amplification
+
+Observation
+-----------
+
+Three data types in Ceph
+- Log (e.g., pglog)
+  - Can be deleted after checkpoint
+  - The best method to store this data is circular log (WAL)
```

Comment:
I think you are confusing the ObjectStore level WAL/journal that exists in each ObjectStore implementation with the osd level PGLog.  You don't control the PGLog entry lifetime and cannot assume that there is any "checkpoint" you have access to.